packages/node/src/lib/construct-payload.ts

```diff
@@ -1,9 +1,29 @@
-const url = require('url');
-const processRequest = require('./process-request');
-const processResponse = require('./process-response');
-const { name, version } = require('../package.json');
+import url from 'url';
+import processRequest from './process-request';
+import processResponse from './process-response';
+import { name, version } from '../../package.json';
+import { Options } from './express-middleware';
 
-module.exports = (req, res, groupingFn, options = {}, { logId, startedDateTime }) => {
+export interface GroupingFunction {
+  (req, res): {
+    // @todo: document this
+    apiKey: string;
+    // @deprecated use apiKey instead
+    id: string;
+    // @todo: document this
+    label: string;
+    // @todo: document this
+    email: string;
+  };
+}
+
+export function constructPayload(
+  req,
+  res,
+  groupingFn: GroupingFunction,
+  options: Options = {},
+  { logId, startedDateTime }
+) {
   const group = groupingFn(req, res);
 
   // if apiKey is specified use it as if it were passed in as the id
```

Comment:
For the lines immediately following this, we transform apiKey to id.
Do we support apiKey server side? or do we always need to perform this translation. It would be great to remove this logic from all the clients and reduce build complexity.

@gratcliff  @ilias-t

Reply:
[Here's the extent](https://github.com/readmeio/readme-metrics-api/blob/next/src/lib/get-group.ts#L1) of our `group` transformation server-side, [called here](https://github.com/readmeio/readme-metrics-api/blob/next/src/routes/v1/request.ts#L74).

It's not something we currently do, but could definitely support going forward.

Reply:
cool, I'll create a ticket. It could probably still persist to groupId, just handle the mapping server side instead of in the client!

Reply:
https://linear.app/readme-io/issue/RM-2141/accept-the-apikey-grouping-parameter